packages/origin-247-transfer/src/batch/queueThrottle.ts

```diff
@@ -0,0 +1,34 @@
+import { Subject, throttleTime, asyncScheduler, concatMap, from, tap, startWith } from 'rxjs';
+
+/**
+ * Ignore all calls for @time, then proceed (and repeat) - at least one call will happen after @time.
+ *
+ * It also queues promises (so they don't run in parallel) - if multiple calls arrive
+ * despite throttling, they will be executed one by one.
+ *
+ * It also makes initial call automatically.
+ *
+ * @param cb - callback that should return Promise
+ * @param throttleSeconds - throttling time in seconds
+ * @returns - function to call to "add to queue"
+ */
+export const queueThrottle = <T>(cb: () => Promise<unknown>, throttleSeconds: number) => {
+    const subject = new Subject<T>();
+
+    subject
+        .pipe(
+            // Initializing part
+            startWith(true),
```

Comment:
Because of this function, even if application is restarted, all services will attempt to process some batches (even if they did not receive any request to do so)